pysb/simulator/base.py

```diff
@@ -0,0 +1,329 @@
+        Verbosity flag passed to the constructor.
+    model : pysb.Model
+        Model passed to the constructor.
+    tspan : vector-like
+        Time values passed to the constructor.
+    tout: numpy.ndarray
+        Time points returned by the simulator (may be different from ``tspan``
+        if simulation is interrupted for some reason).
+
+    Notes
+    -----
+    If ``tspan`` is not defined, it may be defined in the call to the
+    ``run`` method.
+
+    The dimensionality of ``tout`` depends on whether a single simulation
+    or multiple simulations are run.
+
+    The dimensionalities of ``y``, ``yobs``, ``yobs_view``, ``yexpr``, and
+    ``yexpr_view`` depend on the number of simulations being run as well
+    as on the type of simulation, i.e., spatial vs. non-spatial.
+
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def __init__(self, model, verbose=False, **kwargs):
+        self.model = model
+        self.verbose = verbose
+        self.tout = None
+        self._y = None
+        self._yobs = None
+        self._yobs_view = None
+        self._yexpr = None
+        self._yexpr_view = None
+        self._yfull = None
+        # Per-run initial conditions/parameter/tspan override
+        self.tspan = kwargs.get('tspan', None)
+        self._initials = None
+        self.initials = kwargs.get('initials', None)
+        self._params = None
+        self.param_values = kwargs.get('param_values', None)
+
+    def _clear_outputs(self):
+        """
+        Resets simulation output
+        """
+        self._y = None
+        self._yobs = None
+        self._yobs_view = None
+        self._yexpr = None
+        self._yexpr_view = None
+        self._yfull = None
+
+    @property
+    def initials(self):
+        if self._initials is not None:
+            return self._initials
+        else:
+            return self.initials_list
+
+    @initials.setter
+    def initials(self, new_initials):
+        if new_initials is None:
+            self._initials = None
+        else:
+            # check if y0 is a dict, and if so validate the keys
+            # (ComplexPatterns)
+            if isinstance(new_initials, dict):
+                for cplx_pat, val in new_initials.items():
+                    if not isinstance(cplx_pat, (MonomerPattern,
+                                                 ComplexPattern)):
+                        raise SimulatorException('Dictionary key %s is not a '
+                                                 'MonomerPattern or '
+                                                 'ComplexPattern' %
+                                                 repr(cplx_pat))
+                self._initials = new_initials
+            # accept vector of species amounts as an argument
+            elif len(new_initials) != self._y[0].shape[1]:
+                raise ValueError("y0 must be the same length as model.species")
+            elif not isinstance(new_initials, np.ndarray):
```

Comment:
You can replace the last elif and the else with this:
```python
else:
    self._initials = np.array(new_initials, copy=False)
```
That will construct a new array for a non-array, and return a view on an existing array (shares the same memory and avoids a copy).